pepper-apis/src/main/java/org/broadinstitute/ddp/filter/UserAuthCheckFilter.java

```diff
@@ -115,27 +114,33 @@ private void handleTemporaryUser(Request request) {
 
         if (!inWhitelist) {
             LOG.warn("Request '{} {}' is not in temp-user whitelist", requestedMethod, requestedPath);
-            throw halt(401);
+            throw ResponseUtil.haltError(HttpStatus.SC_UNAUTHORIZED,
+                    new ApiError(ErrorCodes.AUTH_CANNOT_BE_DETERMINED, "Request is not in temp-user whitelist"));
         }
 
         UserDto tempUser = TransactionWrapper.withTxn(handle ->
                 handle.attach(JdbiUser.class).findByUserGuid(tempUserGuid));
 
         boolean canAccess = true;
+        String message = null;
         if (tempUser == null) {
             LOG.warn("Could not find temporary user with guid '{}'", tempUserGuid);
+            message = "Could not find temporary user with guid: " + tempUserGuid;
             canAccess = false;
         } else if (!tempUser.isTemporary()) {
             LOG.error("User with guid '{}' is not a temporary user but is used to access"
                     + " temp-user whitelisted path without a token", tempUserGuid);
+            message = "User is not a temporary user";
             canAccess = false;
         } else if (tempUser.isExpired()) {
-            LOG.error("Temporary user with guid '{}' had already expired at time {}ms", tempUserGuid, tempUser.getExpiresAtMillis());
+            LOG.warn("Temporary user with guid '{}' had already expired at time {}ms", tempUserGuid, tempUser.getExpiresAtMillis());
+            message = "Temporary user passed expiration time";
```

Comment:
If I'm understanding, the proposal is to return a generic message about the _class_ of problem and not anything specific. Perhaps we should change these to return something like `...invalid temporary user..` instead?